diff -up tree-1.5.3/tree.c.fixbufsiz tree-1.5.3/tree.c
--- tree-1.5.3/tree.c.fixbufsiz	2014-11-27 16:09:23.487581695 +0000
+++ tree-1.5.3/tree.c	2014-11-27 16:10:35.326947188 +0000
@@ -162,7 +162,7 @@ void listdir(char *, int *, int *, u_lon
 void parse_dir_colors(), printit(char *), free_dir(struct _info **), indent(int maxlevel);
 void saveino(ino_t, dev_t);
 char **split(char *, char *, int *);
-char *gidtoname(int), *uidtoname(int), *do_date(time_t);
+char *gidtoname(gid_t), *uidtoname(uid_t), *do_date(time_t);
 char *gnu_getcwd();
 struct _info **read_dir(char *, int *);
 void html_encode(FILE *, char *), url_encode(FILE *, char *);
@@ -1194,11 +1194,11 @@ char *do_date(time_t t)
   return buf;
 }
 
-char *uidtoname(int uid)
+char *uidtoname(uid_t uid)
 {
   struct xtable *o, *p, *t;
   struct passwd *ent;
-  char ubuf[6];
+  char ubuf[11];
   int uent = HASH(uid);
 
   for(o = p = utable[uent]; p ; p=p->nxt) {
@@ -1210,7 +1210,7 @@ char *uidtoname(int uid)
   t = xmalloc(sizeof(struct xtable));
   if ((ent = getpwuid(uid)) != NULL) t->name = scopy(ent->pw_name);
   else {
-    sprintf(ubuf,"%d",uid);
+    snprintf(ubuf, sizeof (ubuf), "%u", uid);
     t->name = scopy(ubuf);
   }
   t->xid = uid;
@@ -1220,11 +1220,11 @@ char *uidtoname(int uid)
   return t->name;
 }
 
-char *gidtoname(int gid)
+char *gidtoname(gid_t gid)
 {
   struct xtable *o, *p, *t;
   struct group *ent;
-  char gbuf[6];
+  char gbuf[11];
   int gent = HASH(gid);
 
   for(o = p = gtable[gent]; p ; p=p->nxt) {
@@ -1236,7 +1236,7 @@ char *gidtoname(int gid)
   t = xmalloc(sizeof(struct xtable));
   if ((ent = getgrgid(gid)) != NULL) t->name = scopy(ent->gr_name);
   else {
-    sprintf(gbuf,"%d",gid);
+    snprintf(gbuf, sizeof (gbuf), "%u", gid);
     t->name = scopy(gbuf);
   }
   t->xid = gid;
