diff -up tree-1.6.0/Makefile.no-strip tree-1.6.0/Makefile
--- tree-1.6.0/Makefile.no-strip	2011-06-24 15:25:27.000000000 +0100
+++ tree-1.6.0/Makefile	2011-07-04 11:11:17.662036666 +0100
@@ -32,7 +32,7 @@ OBJS=tree.o unix.o html.o xml.o hash.o c
 # Linux defaults:
 #CFLAGS=-ggdb -Wall -DLINUX -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64
 CFLAGS=-O4 -Wall  -DLINUX -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64
-LDFLAGS=-s
+LDFLAGS=
 
 # Uncomment for FreeBSD:
 #CFLAGS=-O2 -Wall -fomit-frame-pointer
@@ -97,7 +97,7 @@ install: tree
 	install -d $(BINDIR)
 	install -d $(MANDIR)
 	if [ -e $(TREE_DEST) ]; then \
-		install -s $(TREE_DEST) $(BINDIR)/$(TREE_DEST); \
+		install $(TREE_DEST) $(BINDIR)/$(TREE_DEST); \
 	fi
 	install doc/$(MAN) $(MANDIR)/$(MAN)
 
